pkg/controller/datavolume-controller.go

```diff
@@ -524,6 +559,94 @@ func (c *DataVolumeController) updateDataVolumeStatus(dataVolume *cdiv1.DataVolu
 	return err
 }
 
+func (c *DataVolumeController) updateProgress(dataVolumeCopy *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) {
+	hasProgress := false
+	_, ok := pvc.Annotations[AnnImportPod]
+	if ok {
+		hasProgress = true
+	}
+	_, ok = pvc.Annotations[AnnCloneRequest]
+	if ok {
+		hasProgress = true
+	}
+	if hasProgress {
+		glog.V(3).Info("Has annotation that can report progress")
+		// Pod will be reporting progress, on port 443, attempt to get it.
+		l, err := labels.Parse("prometheus.kubevirt.io")
+		if err != nil {
+			return
+		}
+		pods, err := c.kubeclientset.CoreV1().Pods(pvc.Namespace).List(metav1.ListOptions{LabelSelector: l.String()})
+		if err != nil {
+			return
+		}
+		for _, pod := range pods.Items {
+			glog.V(3).Info("***Pod owner reference: " + pod.OwnerReferences[0].UID + ", pvc UID: " + pvc.GetUID())
+			if pod.OwnerReferences[0].UID == pvc.GetUID() {
+				glog.V(3).Info("***Pod Name: " + pod.Name)
+				if len(pod.Status.PodIP) == 0 {
+					glog.V(3).Info("No IP address for pod found.")
+					return
+				}
+				c.updateProgressUsingPod(dataVolumeCopy, pvc, &pod)
+				return
+			}
+		}
+	}
+}
+
+func (c *DataVolumeController) updateProgressUsingPod(dataVolumeCopy *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim, pod *corev1.Pod) {
+	httpClient := buildHTTPClient()
+	// Example value: import_progress{ownerUID="b856691e-1038-11e9-a5ab-525500d15501"} 13.45
+	var importRegExp = regexp.MustCompile("progress\\{ownerUID\\=\"" + string(dataVolumeCopy.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
+
+	url := fmt.Sprintf("https://%s:8443/metrics", pod.Status.PodIP)
```

Comment:
This reminds me, should we look at making this port configurable?

Reply:
Well instead of hard coding, I should probably look it up from the service. In theory its already configurable from the service.

Reply:
+1, yeah pulling it from the service is way more clear (for me at least)